packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -692,11 +690,11 @@ class _ModalScopeStatus extends InheritedWidget {
 
 class _ModalScope<T> extends StatefulWidget {
   const _ModalScope({
-    Key key,
+    Key? key,
     this.route,
   }) : super(key: key);
 
-  final ModalRoute<T> route;
+  final ModalRoute<T>? route;
```

Comment:
this can be non-nullable, it's only ever called with `this` as the argument value

Reply:
Done